rust/arrow/src/array/data.rs

```diff
@@ -136,6 +137,84 @@ impl ArrayData {
         &self.null_bitmap
     }
 
+    /// Computes the logical validity bitmap of the array data using the
+    /// parent's array data. The parent should be a list or struct, else
+    /// the logical bitmap of the array is returned unaltered.
+    ///
+    /// Parent data is passed along with the parent's logical bitmap, as
+    /// nested arrays could have a logical bitmap different to the physical
+    /// one on the `ArrayData`.
+    ///
+    /// Safety
+    ///
+    /// As we index into [`ArrayData::child_data`], this function panics if
+    /// array data is not a nested type, as it will not have child data.
+    pub fn child_logical_null_buffer(
+        &self,
+        logical_null_buffer: Option<Buffer>,
+        child_index: usize,
```

Comment:
I would have pass `&ArrayData` instead of `child_index`. This is available in all instances where this is called.